mlflow/utils/file_utils.py

```diff
@@ -967,3 +969,26 @@ def chdir(path: str) -> None:
         yield
     finally:
         os.chdir(cwd)
+
+
+def get_total_file_size(path: str) -> int:
+    """
+    Return the size of all files under given path, including files in subdirectories.
+
+    :param path: The absolute path of a local directory.
+    :return: size in bytes.
+    """
+    if not os.path.exists(path):
+        raise MlflowException(
+            message=f"The given {path} does not exist.", error_code=INVALID_PARAMETER_VALUE
+        )
+    if not os.path.isdir(path):
+        raise MlflowException(
+            message=f"The given {path} is not a directory.", error_code=INVALID_PARAMETER_VALUE
+        )
+
+    total_size = 0
+    for cur_path, dirs, files in os.walk(path):
+        full_paths = [os.path.join(cur_path, file) for file in files]
+        total_size += sum([os.path.getsize(file) for file in full_paths])
+    return total_size
```

Comment:
/ai make this a one-liner using pathlib.

Reply:
/ai Nice. Does `rglob` only yield a file?

Reply:
+1 to rglob with wildcard. Pathlib's traversal algorithm is far more sophisticated and performant than os.walk(). 
@wenfeiy-db can we implement this recommended change? :)

Reply:
Ahh I assumed the traversal will be similar - let me try changing it to use pathlib

Reply:
@BenWilson2 actually, can you explain how pathlib's traversal is more performant? It seems `pathlib` operations are normally slower than `os` operations ([source1](https://switowski.com/blog/pathlib/), [source2](https://treyhunner.com/2019/01/no-really-pathlib-is-great/)).

I can let `get_total_file_size` accept both str and pathlib.Path, but I'd like to understand what're other benefits (in additional to it's an one-liner) of using pathlib's rglob instead of os.walk?

Reply:
I don't think pathlib is faster. I just wanted to make this change to simplify the code.